packages/utils/src/dal/mikro-orm/mikro-orm-repository.ts

```diff
@@ -420,6 +435,329 @@ export function mikroOrmBaseRepositoryFactory<T extends object = object>(
       // TODO return the all, created, updated entities
       return upsertedEntities
     }
+
+    // UpsertWithReplace does several things to simplify module implementation.
+    // For each entry of your base entity, it will go through all relations, and it will do a diff between what is passed and what is in the database.
+    // For each relation, it create new entries (without an ID), it will associate existing entries (with only an ID), and it will update existing entries (with an ID and other fields).
+    // Finally, it will delete the relation entries that were omitted in the new data.
+    // The response is a POJO of the data that was written to the DB, including all new IDs. The order is preserved with the input.
+    // Limitations: We expect that IDs are used as primary keys, and we don't support composite keys. Also, we only support 1-level depth of upserts
```

Comment:
q: would a many to many with a pivot entity more than the join keys be supported here? Might be worth to add that in as a limitation as we have a couple of models like that

Reply:
in theory mikro orm would still resolve the join columns internally, we can eventually add a test to validate that but it should work

Reply:
@riqwan good point, it's not supported now and I can add it as a limitation